worksheet_D/TicTacToe.py

```diff
@@ -0,0 +1,114 @@
+class OxoBoard:
+
+    def __init__(self,x,y):
+        self.x = x
+        self.y = y
+        self.board = {(0,0):0,(0,1):0,(0,2):0,(1,0):0,(2,0):0,(1,1):0,(2,2):0,(1,2):0,(2,1):0}

```

Comment:
OK, but might be neater to initialise the dictionary with a loop e.g.
```
self.board = {}
for x in xrange(3):
    for y in xrange(3):
        self.board[x, y] = 0
```

Also, the order of keys here is a bit strange and makes it difficult to see that all keys are present and correct